community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/TicketedProcessing.java

```diff
@@ -86,7 +86,7 @@ public boolean test( long ticket )
         public boolean test( long ticket )
         {
             long queued = submittedTicket.get() - processedTicket.get();
-            return queued <= executor.processors( 0 );
+            return queued <= executor.processors( 0 ) | executor.isShutdown();
```

Comment:
I forgot and trying to reproduce failed... all I know is that there was a 50/50 chance normal shutdown (or panic, dunno) hung the importer and that the added condition removed that issue.